### DISCUSSION TITLE
[Ai-deck]Unable to flash GAP8 via Olimex-ARM-USB_TINY-H JTAG

### DISCUSSION THREAD

**Discussion by pulsar666** [bitcraze/discussions#381] [Q&A]
Hello,
I'm trying trying to upload code to my Aideck's GAP8 via the Olimex JTAG. I have provided udev permissions by creating a .rules file with proper pid and vid. I'm able to upload code to the ESP32 via the same JTAG through EspressIF IDE. However, when I run the following docker command:

`docker run --rm -v ${PWD}:/module --device /dev/ttyUSB0 --privileged -P bitcraze/aideck tools/build/make-example /gap8_core_benchmark flash`

I get the following response:
gapy --target=ai_deck --platform=board --work-dir=/module/gap8_core_benchmark/BUILD/GAP8_V2/GCC_RISCV_FREERTOS --config-ini=/module/gap8_core_benchmark/config.ini    run --flash --force --binary=/module/gap8_core_benchmark/BUILD/GAP8_V2/GCC_RISCV_FREERTOS/gap8_benchmark 
Open On-Chip Debugger 0.10.0+dev-gb84d97ec4 (2022-04-20-14:26)
Licensed under GNU GPL v2
For bug reports, read
	http://openocd.org/doc/doxygen/bugs.html
debug_level: 0

DEPRECATED! use 'adapter speed' not 'adapter_khz'
DEPRECATED! use 'adapter driver' not 'interface'
TARGET create
GAP8 INIT TARGET
Error: no device found
Error: unable to open ftdi device with vid 15ba, pid 002a, description '**Olimex OpenOCD JTAG ARM-USB-TINY-H**', serial '*' at bus location '*'

Flashing image with command:
gap8-openocd -d0 -c "gdb_port disabled; telnet_port disabled; tcl_port disabled" -c "script interface/ftdi/olimex-arm-usb-tiny-h.cfg; script target/gap8revb.tcl; script tcl/flash_image.tcl; script tcl/jtag_boot.tcl; gap_flash_raw_hyper /module/gap8_core_benchmark/BUILD/GAP8_V2/GCC_RISCV_FREERTOS/target.board.devices.flash.img 49184 /gap_sdk/utils/openocd_tools; exit;"
Traceback (most recent call last):
  File "/gap_sdk/utils/gapy/gapy", line 162, in <module>
    main()
  File "/gap_sdk/utils/gapy/gapy", line 152, in main
    operationFunc(args, config, system)
  File "/gap_sdk/utils/gapy/run.py", line 184, in operationFunc
    raise RuntimeError('Runner has failed with value: %d' % status)
RuntimeError: Runner has failed with value: -1
make: *** [/gap_sdk/rtos/freeRTOS/vendors/gwt/rules/freeRTOS_rules.mk:404: flash] Error 1

As you can see, the JTAG USB descriptor name that its pulling from the " interface/ftdi/olimex-arm-usb-tiny-h.cfg" file seems to be "Olimex OpenOCD JTAG ARM-USB-TINY-H". However, when I do a lsusb, I get the following: 
<img width="416" alt="image" src="https://user-images.githubusercontent.com/19473511/200773606-535db31c-505d-432b-97e3-75f9d01f87cb.png">

Is this the cause for the error? If so, where can I find this configuration file (because it "seems" to me that it's not using the file on my local openccd installation) so that I may change the USB descriptor name in it. 
If this not the issue, then what else could it be?

Please help!

**Comment by evoggy** [comment]
Hi,

I tested this a bit and I get the same error as you when the dongle is not connected. So it seems as if the dongle is not accessible from within docker. I also get the same error when having the adapter connected but not including the ```--privileged```.
```
gapy --target=ai_deck --platform=board --work-dir=/module/examples/other/wifi-img-streamer/BUILD/GAP8_V2/GCC_RISCV_FREERTOS --config-ini=/module/examples/other/wifi-img-streamer/config.ini    run --flash --force --binary=/module/examples/other/wifi-img-streamer/BUILD/GAP8_V2/GCC_RISCV_FREERTOS/wifi-img-streamer 
Open On-Chip Debugger 0.10.0+dev-gb84d97ec4 (2022-04-20-14:26)
Licensed under GNU GPL v2
For bug reports, read
	http://openocd.org/doc/doxygen/bugs.html
debug_level: 0

DEPRECATED! use 'adapter speed' not 'adapter_khz'
DEPRECATED! use 'adapter driver' not 'interface'
TARGET create
GAP8 INIT TARGET
Error: no device found
Error: unable to open ftdi device with vid 15ba, pid 002a, description 'Olimex OpenOCD JTAG ARM-USB-TINY-H', serial '*' at bus location '*'

Flashing image with command:
gap8-openocd -d0 -c "gdb_port disabled; telnet_port disabled; tcl_port disabled" -c "script interface/ftdi/olimex-arm-usb-tiny-h.cfg; script target/gap8revb.tcl; script tcl/flash_image.tcl; script tcl/jtag_boot.tcl; gap_flash_raw_hyper /module/examples/other/wifi-img-streamer/BUILD/GAP8_V2/GCC_RISCV_FREERTOS/target.board.devices.flash.img 63520 /gap_sdk/utils/openocd_tools; exit;"
Traceback (most recent call last):
  File "/gap_sdk/utils/gapy/gapy", line 162, in <module>
    main()
  File "/gap_sdk/utils/gapy/gapy", line 152, in main
    operationFunc(args, config, system)
  File "/gap_sdk/utils/gapy/run.py", line 184, in operationFunc
    raise RuntimeError('Runner has failed with value: %d' % status)
RuntimeError: Runner has failed with value: -1
make: *** [/gap_sdk/rtos/freeRTOS/vendors/gwt/rules/freeRTOS_rules.mk:404: flash] Error 1
```

Can you please check what version of the container you have? Please run the following:
```
$ docker images | grep bitcraze/aideck
bitcraze/aideck                                         4.8.0.2          2e6220ef1f0f   6 months ago    5.15GB
```
But when you're uploading the ESP firmware, are you using the docker container or the native install?

Could also run the following command, maybe the dongle is not accessable as ```ttyUSB0```?
```
$ ls /dev/ttyUSB*
/dev/ttyUSB0
```

**Reply by pulsar666** [reply]
Thanks for the reply!
1. "But when you're uploading the ESP firmware, are you using the docker container or the native install?" ---
Native install (EspressIf IDE) not docker

2. "Could also run the following command, maybe the dongle is not accessable as ttyUSB0?

$ ls /dev/ttyUSB*
/dev/ttyUSB0" ---

Returns the same thing: /dev/ttyUSB0

3. bitcraze/ai-deck container version is showing "latest" and not printing a particular version number as in your case.
<img width="370" alt="image" src="https://user-images.githubusercontent.com/19473511/202152609-8b6af301-e51b-413d-9367-98800471a9c2.png">

I have followed the instructions in the following link to set up the container: 
[https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/#setting-up-docker-and-the-autotiler](url)

**Reply by evoggy** [reply]
Everything looks correct. Are you on native Ubuntu or are you running in a VM or WSL?

**Reply by pulsar666** [reply]
Native Ubuntu 20.04 Jammy

**Reply by evoggy** [reply]
I'm not sure what the issue is here, from what I can see it should be working. There's also the solution to install the SDK natively (or at least the openOCD part). Here's more information on how to do it: https://github.com/GreenWaves-Technologies/gap_sdk/tree/release-v4.8.0#getting-started-with-the-gap-sdk

**Reply by boomer319** [reply]
Don't forget to do either `export GAPY_OPENOCD_CABLE=interface/ftdi/olimex-arm-usb-tiny-h.cfg` or `export GAPY_OPENOCD_CABLE=interface/jlink.cfg` based on what jtag adapter you use. I was able to clear your initial error using this command. This part of the "[getting started guide](https://github.com/GreenWaves-Technologies/gap_sdk/tree/release-v4.8.0#getting-started-with-the-gap-sdk)" is unfortunately quite low on the page.
